### ISSUE TITLE
False Positive "Variable ... does not need to be mutable." Within Juniper graphql_object

### ISSUE THREAD

**Issue by nicholasgrose** [intellij-rust/intellij-rust#10529]
## Environment

* **IntelliJ Rust plugin version:**
* 0.4.195.5408-231
* **Rust toolchain version:**
* 1.72.0-nightly
* **IDE name and version:**
* CLion 2023.1.3
* **Operating system:**
* Fedora 38 (GNOME)

## Problem description
Mutable variables in the context of a #[graphql_object] macro from Juniper (0.15.11, in my case) have a false positive inspection claiming they don't need to be mutable when they must be to be used:
![image](https://github.com/intellij-rust/intellij-rust/assets/35816886/9469dd99-aa49-42ca-b9ac-7c67732bc9d8)

## Steps to reproduce
1. Open up a file containing mutable variables passed into functions in a Juniper #[graphql_object], like the following:
```rust
pub struct TestContext;

impl juniper::Context for TestContext {}

pub struct TestQuery;

#[graphql_object(context = TestContext)]
impl TestQuery {
    async fn name(
        _context: &TestContext,
    ) -> &'static str {
        let mut thing = "".to_string();

        do_something(&mut thing);

        ""
    }
}

fn do_something(a: &mut str) {
    println!("{}", a);
}
```
2. Observe that the mutable variable "thing" gets highlighted with the false positive.

NOTE: This also seems to occur in closures, too, within the Juniper resolvers.

**Comment by nicholasgrose** [comment]
This is awfully strange. I do not seem to be observing this anymore. Even after trying to toggle on and off all the experimental features and reloading the project in both stable and nightly, and I no longer see this. I wonder if something went weird because of the nightly update (like from Clippy, since I use that):
![image](https://github.com/intellij-rust/intellij-rust/assets/35816886/61dd24ed-8401-4533-b1fc-9e9066baffa1)
 
I suppose it is also possible it was something going weird somewhere, and I just can't observe it anymore after IDE restart. It's hard to tell, so I'll close this issue because it is apparently not reproducible.
